lib/tasks_private/terraform/provision.rake

```diff
@@ -0,0 +1,84 @@
+require "json"
+require 'yaml'
+require 'logger'
+require 'ruby-terraform'
+
+namespace :provision do
+  desc "check terraform is installed"
+  task :check => :environment do
+    system("command -v terraform > /dev/null") or raise("install terraform to provision powervc vm")
+  end
+
+  desc "provision a vm"
+  task :apply => :check do
+    cdir = ENV['cwd'] || Dir.pwd
+    tdir = __dir__
+    config = setup(cdir, tdir)
+    plan(config)
+    apply(cdir, tdir)
+  end
+
+  desc "delete a vm"
+  task :destroy => :check do
+    cdir = ENV['cwd'] || Dir.pwd
+    tdir = __dir__
+    config = setup(cdir, tdir)
+    destroy(cdir, config)
+  end
+
+  def setup(cdir, tdir)
+    config = YAML.safe_load(File.read(File.join(cdir, "config/secrets.yml")))
```

Comment:
Super minor but this can be `YAML.load_file()` and if you make `cdir` a `Pathname` like `cdir = Pathanme.new(ENV['cwd'] || Dir.pwd)` the whole like can be `YAML.safe_load(cdir.join("config/secrets.yml"))`